blockchain/projects.go

```diff
@@ -232,16 +224,33 @@ func (p *Projects) DeployContract(
 	if err != nil {
 		return nil, err
 	}
+
 	if _, ok := flowAccount.Contracts[contractName]; ok {
 		// A contract with this name has already been deployed to this account
-		// TODO: change logic to just re-deploy this contract instead of resetting the state
-		var proj model.Project
-		err := p.store.GetProject(projectID, &proj)
+		// Remove previous contract from the account
+		result, tx, err := em.removeContract(address.ToFlowAddress(), contractName)
+		if err != nil {
+			return nil, err
+		}
+		if result.Error != nil {
+			return nil, result.Error
+		}
+
+		// Insert transaction execution of contract removal into db
+		exe := model.TransactionExecutionFromFlow(projectID, result, tx)
+		err = p.store.InsertTransactionExecution(exe)
+		if err != nil {
+			return nil, err
+		}
+
+		// Remove contract deployment from db
+		var deployment model.ContractDeployment
+		err = p.store.GetContractDeploymentByName(projectID, address, contractName, &deployment)
```

Comment:
Do you use this method anywhere else? otherwise wouldn't it be better to just have `DeleteContractDeploymentByName`?
Also dangerous thing to do out of db transaction, I created a ticket: https://github.com/onflow/flow-playground-api/issues/131